backends/tfhe-cuda-backend/cuda/src/pbs/programmable_bootstrap_classic_128.cu

```diff
@@ -0,0 +1,205 @@
+#include "programmable_bootstrap_classic_128.cuh"
+
+/*
+ * This scratch function allocates the necessary amount of data on the GPU for
+ * the PBS on 128 bits inputs, into `buffer`. It also configures SM options on
+ * the GPU in case FULLSM or PARTIALSM mode is going to be used.
+ */
+void scratch_cuda_programmable_bootstrap_128(
+    void *stream, uint32_t gpu_index, int8_t **pbs_buffer,
+    uint32_t glwe_dimension, uint32_t polynomial_size, uint32_t level_count,
+    uint32_t input_lwe_ciphertext_count, bool allocate_gpu_memory) {
+
+  auto buffer = (pbs_buffer_128<CLASSICAL> **)pbs_buffer;
+  switch (polynomial_size) {
+  case 256:
+    scratch_programmable_bootstrap_128<AmortizedDegree<256>>(
+        static_cast<cudaStream_t>(stream), gpu_index, buffer, glwe_dimension,
+        polynomial_size, level_count, input_lwe_ciphertext_count,
+        allocate_gpu_memory);
+    break;
+  case 512:
+    scratch_programmable_bootstrap_128<AmortizedDegree<512>>(
+        static_cast<cudaStream_t>(stream), gpu_index, buffer, glwe_dimension,
+        polynomial_size, level_count, input_lwe_ciphertext_count,
+        allocate_gpu_memory);
+    break;
+  case 1024:
+    scratch_programmable_bootstrap_128<AmortizedDegree<1024>>(
+        static_cast<cudaStream_t>(stream), gpu_index, buffer, glwe_dimension,
+        polynomial_size, level_count, input_lwe_ciphertext_count,
+        allocate_gpu_memory);
+    break;
+  case 2048:
+    scratch_programmable_bootstrap_128<AmortizedDegree<2048>>(
+        static_cast<cudaStream_t>(stream), gpu_index, buffer, glwe_dimension,
+        polynomial_size, level_count, input_lwe_ciphertext_count,
+        allocate_gpu_memory);
+    break;
+  case 4096:
+    scratch_programmable_bootstrap_128<AmortizedDegree<4096>>(
+        static_cast<cudaStream_t>(stream), gpu_index, buffer, glwe_dimension,
+        polynomial_size, level_count, input_lwe_ciphertext_count,
+        allocate_gpu_memory);
+    break;
+  default:
+    PANIC("Cuda error (classical PBS): unsupported polynomial size. "
+          "Supported N's are powers of two"
+          " in the interval [256..4096].")
+  }
+}
+
+template <typename Torus>
+void executor_cuda_programmable_bootstrap_lwe_ciphertext_vector_128(
```

Comment:
Usually we don't have an `executor_` functions, only `cuda_programmable_bootstrap_lwe_ciphertext_vector_128` in this case. You could directly paste the content of this function there, no?

Reply:
I would keep it like this,  when we add new variants of 128 bit PBS like CG, TBC or even multi_bit variants there will be another switch for them and it will be a big amount of code with nested switches without executor.

Reply:
Ok :+1: